Gems/Multiplayer/Code/Source/Components/NetworkDebugPlayerIdComponent.cpp

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <Multiplayer/Components/NetworkDebugPlayerIdComponent.h>
+
+#include <Atom/RPI.Public/ViewportContextBus.h>
+#include <AzCore/Serialization/SerializeContext.h>
+#include <AzFramework/Viewport/ViewportScreen.h>
+#include <AzNetworking/Framework/INetworking.h>
+
+namespace Multiplayer
+{
+    AZ_CVAR(bool, cl_debugPlayerIdRender, true, nullptr, AZ::ConsoleFunctorFlags::DontReplicate, "Whether NetworkDebugPlayerIdComponent should render the player id and connection count on-screen.");
+
+    void NetworkDebugPlayerIdComponent::Reflect(AZ::ReflectContext* context)
+    {
+        if (const auto serializeContext = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serializeContext->Class<NetworkDebugPlayerIdComponent, NetworkDebugPlayerIdComponentBase>()
+                ->Version(1)
+                ->Field("translationOffset", &NetworkDebugPlayerIdComponent::m_translationOffset)
+                ->Field("scale", &NetworkDebugPlayerIdComponent::m_fontScale)
+                ->Field("color", &NetworkDebugPlayerIdComponent::m_fontColor)
+            ;
+
+            if (AZ::EditContext* editContext = serializeContext->GetEditContext())
+            {
+                editContext->Class<NetworkDebugPlayerIdComponent>("Network Debug Player ID", "Renders the player id as debug text over network players.")
+                    ->ClassElement(AZ::Edit::ClassElements::EditorData, "")
+                        ->Attribute(AZ::Edit::Attributes::Category, "Multiplayer")
+                        ->Attribute(AZ::Edit::Attributes::AppearsInAddComponentMenu, AZ_CRC_CE("Game"))
+                    ->DataElement(AZ::Edit::UIHandlers::Default, &NetworkDebugPlayerIdComponent::m_translationOffset, "Translation Offset", "The world-space offset from the player position to render the debug text.")
+                    ->DataElement(AZ::Edit::UIHandlers::Default, &NetworkDebugPlayerIdComponent::m_fontScale, "Font Scale", "Apply a scale to the default debug font rendering size.")
```

Comment:
what _is_ the default debug font rendering size?

Reply:
"This big" _*closes one eye and looks through pinched fingers_

Oh geez, I wish I knew. It looks like 12pt font, but I'd be surprised if it's using a standard [font size unit](https://m2.material.io/design/typography/the-type-system.html#type-scale)

Reply:
https://github.com/o3de/o3de.org/issues/2249